File: FrankGUI/MainViewController.m
//
//  MainViewController.m
//  FrankGUI
//
//  Created by Oleksiy Radyvanyuk on 18/05/15.
//  Copyright (c) 2015 Oleksiy Radyvanyuk. All rights reserved.
//

#import "MainViewController.h"
#import "FrankGUI-Swift.h"
#import "AppDelegate.h"

@interface MainViewController ()

@property (nonatomic, weak) Settings *settings;

@end

@implementation MainViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    // prepare Settings object for being used
    self.settings = [(AppDelegate *)[[NSApplication sharedApplication] delegate] settings];
}

- (void)viewDidAppear
{
    [super viewDidAppear];

    switch (self.settings.tabChoice) {
        case TabChoiceRunner:
            self.selectedTabViewItemIndex = 0;
            break;
            
        case TabChoiceSettings:
            self.selectedTabViewItemIndex = [self.tabViewItems count] - 1;
            break;
            
        default:
            break;
    }

    self.tabView.delegate = self;
}

- (void)tabView:(NSTabView *)tabView didSelectTabViewItem:(NSTabViewItem *)tabViewItem
{
    [super tabView:tabView didSelectTabViewItem:tabViewItem];

    if (tabView == self.tabView)
    {
        NSInteger index = [tabView indexOfTabViewItem:tabViewItem];
        if (index == 0)
        {
            self.settings.tabChoice = TabChoiceRunner;
        }
        else if (index == [tabView numberOfTabViewItems] - 1)
        {
            self.settings.tabChoice = TabChoiceSettings;
        }
    }
}

@end

File: FrankGUI/MainViewController.swift
//
//  MainViewController.swift
//  FrankGUI
//
//  Created by Oleksiy Radyvanyuk on 24/06/15.
//  Copyright © 2015 Oleksiy Radyvanyuk. All rights reserved.
//

import Cocoa

final class MainViewController : NSTabViewController
{
    private var settings: Settings?

    required init?(coder: NSCoder) {
        super.init(coder: coder)
        settings = nil
    }

    override func viewDidLoad()
    {
        super.viewDidLoad()

        let delegate = NSApplication.sharedApplication().delegate as! AppDelegate
        settings = delegate.settings
    }
    
    override func viewDidAppear()
    {
        super.viewDidAppear()

        switch settings!.tabChoice
        {
        case .Runner:
            selectedTabViewItemIndex = 0
        case .Settings:
            selectedTabViewItemIndex = tabViewItems.count - 1
        default:
            print("Wut?")
        }
        
        tabView.delegate = self
    }
    
    override func tabView(tabView: NSTabView, didSelectTabViewItem tabViewItem: NSTabViewItem?)
    {
        super.tabView(tabView, didSelectTabViewItem: tabViewItem)
        
        if tabView == self.tabView
        {
            let index = tabView.indexOfTabViewItem(tabViewItem!)
            if index == 0
            {
                settings!.tabChoice = .Runner
            }
            else if index == tabViewItems.count - 1
            {
                settings!.tabChoice = .Settings
            }
        }
    }
}
